graphbolt/include/graphbolt/csc_sampling_graph.h

```diff
@@ -222,12 +224,8 @@ class CSCSamplingGraph : public torch::CustomClassHolder {
       const std::string& shared_memory_name);
 
  private:
-  template <SamplerType S>
```

Comment:
Using std::function to dispatch the appropriate function is going to be slower than the current code. I would have preferred the use of templates. Compared to templates, the advantage is the smaller generated code size. What happens when we use std::function is the dispatch happens at runtime and the compiler can't optimize PickFn and the code calling PickFn together. It becomes slower than simply using if expressions to call one function or the other. You can see this link: https://stackoverflow.com/questions/67615330/why-stdfunction-is-too-slow-is-cpu-cant-utilize-instruction-reordering

However, if PickFn takes a very long time, this overhead can simply be ignored. In our case, it is called `nodes.size()` times. I personally wouldn't want to pay this overhead `nodes.size()` times.

The solution would be to employ the use of templates, depending on whether `fanouts.size() > 1`, we can dispatch the appropriate `PickFn` at compile time. For this to work, instead of defining `PickFn` globally, we need to make it a template parameter.

Reply:
That's a good suggestion! We'll give it a try to see if there are any unforeseen obstacles. If everything goes smoothly, we can adopt this solution without any issues.